### PR TITLE
[xpu][fix][inductor] fix ut SDPAPatternRewriterGpuDynamicTests.test_skip_non_tf32

### PR DESCRIPTION
**Pull request by jianyizh**
cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

File: test/inductor/test_fused_attention.py [hunk 1]
```diff
@@ -1253,10 +1253,10 @@ def forward(self, x1, attn_mask):
...
                         t5 = t4.matmul(v)
                         return t5
-
-                func = Model().to("cuda")
-                x1 = torch.randn(1, 16, 64, 64, device="cuda")
-                attn_mask = torch.zeros(1, 1, 16, 16, device="cuda")
+                device = torch.accelerator.current_accelerator().type
+                func = Model().to(device)
+                x1 = torch.randn(1, 16, 64, 64, device=device)
+                attn_mask = torch.zeros(1, 1, 16, 16, device=device)
```
Comment (guangyey) [inline test/inductor/test_fused_attention.py]:
```suggestion
                func = Model().to(self.device)
                x1 = torch.randn(1, 16, 64, 64, device=self.device)
                attn_mask = torch.zeros(1, 1, 16, 16, device=self.device)
```

File: torch/_inductor/fx_passes/fuse_attention.py [hunk 2]
```diff
@@ -731,7 +731,7 @@ def _sfdp_params_check(match):
         return False
     # fused kernels use tf32
     if (
-        query.device.type == "cuda"
+        query.device.type != "cpu"
         and query.dtype == torch.float32
```
Comment (guangyey) [inline torch/_inductor/fx_passes/fuse_attention.py:741]: Per my understanding, when dtype is `float32`, SDPA may introduce accuracy issues. On CPU, no additional handling is currently applied. On CUDA, https://github.com/pytorch/pytorch/pull/172951 distinguishes two cases:

- If `torch.backends.cuda.matmul.allow_tf32` is disabled, Inductor does not apply the SDPA pattern.

- If `torch.backends.cuda.matmul.allow_tf32` is enabled, the SDPA pattern is applied and the potential accuracy impact is considered acceptable.

I found that sdpa accuary mentoned in https://github.com/pytorch/pytorch/issues/151098 existed on XPU as well. I suppose the reason is that low accuracy was adopted on XPU similar to CUDA. If yes, I think we could follow the same behavior for XPU: when `torch.backends.mkldnn.allow_tf32` is disabled, the SDPA pattern should also not be applied on XPU.

**Review by guangyey — DISMISSED** [review]

File: test/inductor/test_fused_attention.py [hunk 3]
```diff
@@ -1232,40 +1233,46 @@ class SDPAPatternRewriterGpuTests(TestSDPAPatternRewriterTemplate):
 
         @skipIfXpu(msg="FIXME: ENable for XPU")
```
Comment (guangyey) [inline test/inductor/test_fused_attention.py]:
```suggestion
```
